src/event.cc

```diff
@@ -258,21 +261,28 @@ EventScheduler::clean()
 }
 
 void
-EventScheduler::dump(Packable *out)
+EventScheduler::dump(std::ostream &yaml)
 {
+    static const SBuf indent("  ",2);
+
     if (last_event_ran)
-        out->appendf("Last event to run: %s\n\n", last_event_ran);
+        yaml << "last event to run: " << last_event_ran << "\n";
 
-    out->appendf("%-25s\t%-15s\t%s\t%s\n",
-                 "Operation",
-                 "Next Execution",
-                 "Weight",
-                 "Callback Valid?");
+    const auto savedFlags = yaml.flags();
 
+    AtMostOnce header("scheduled events:\n");
     for (auto *e = tasks; e; e = e->next) {
-        out->appendf("%-25s\t%0.3f sec\t%5d\t %s\n",
-                     e->name, (e->when ? e->when - current_dtime : 0), e->weight,
-                     (e->arg && e->cbdata) ? cbdataReferenceValid(e->arg) ? "yes" : "no" : "N/A");
+        yaml << header <<
+             indent << "- operation: " << e->name << '\n' <<
+             indent << indent << "secs to next execution: " << std::setprecision(3) << std::fixed << (e->when? e->when - current_dtime : 0) << '\n';
```

Comment:
Please avoid unnecessary abbreviations, especially non-standard ones.

```suggestion
             indent << indent << "seconds to next execution: " << std::setprecision(3) << std::fixed << (e->when ? e->when - current_dtime : 0) << '\n';
```

Many of these events wait for _hours_. I recommend removing "seconds" from the item key so that we can easily change it later, when we improve support for time units:

```suggestion
             indent << indent << "next execution in: " << std::setprecision(3) << std::fixed << (e->when ? e->when - current_dtime : 0) << " seconds\n";
```

... or something like that.

Finally, many of these events do _not_ repeat so "next" is quite misleading here. I would rephrase further to avoid the (often wrong) implication (e.g., "runs in ... seconds" or "scheduled to run in ... seconds").

Reply:
`... seconds`, from the YAML schema perspective, turns the field from a number into a string, requiring additional work for automations.
In fact, in an ideal world I would specify the unit in a comment for the benefit of humans only: specifying it in the key means that if we change time unit the key changes

Reply:
we can use an inline comment:  `next execution in: ... # seconds`

Reply:
> `... seconds`, from the YAML schema perspective, turns the field from a number into a string, requiring additional work for automations.

I strongly disagree with the implication that consuming unitless numbers (or dealing with field key changes) is overall better "for automations" than consuming numbers with units (because the latter requires "additional work"). And since lots of Squid report fields do use units of various kinds, and using units often makes reports more readable for humans, that "additional work" will most likely be required _anyway_ and will be amortized across various consumed reports.

> In fact, in an ideal world I would specify the unit in a comment for the benefit of humans only:

Our ideals differ: Ideally, as a consumer application developer, I would want units to be correctly processed by my code rather than hidden from it in a YAML comment, especially if units change based on the reported value (to assist human readers).

> specifying it in the key means that if we change time unit the key changes

Agreed. The unit should not go into the field key, especially when the unit should change based on the reported value (for human benefit). That is why I explicitly [recommended](https://github.com/squid-cache/squid/pull/1792#discussion_r1583684955) removing PR-added seconds from the field key.